dialogflow-read.js

```diff
@@ -0,0 +1,41 @@
+'use strict';
+function Read() {
+    this.Read=function(intentId) {
+        const dialogflow = require('dialogflow');
+        const intentsClient = new dialogflow.IntentsClient();
+        const request = {
+            intentView: 'INTENT_VIEW_FULL',
+            name: intentId
+        };
+        intentsClient
+            .getIntent(request)
+            .then(responses => {
+            console.log(responses[0]);
+        var utterance = {
+            "displayName" : "TurnOn",
+            "trainingPhrases":[]
+        }
+        responses[0].trainingPhrases.map(function(item,index){
```

Comment:
`map` on `Array` has an return value, which is a new array. So you can write this part of code like this:
```javascript
utterance.trainingPhrases = responses[0].trainingPhrases.map(item => {
  const parts = item.parts.map(item1 => {
    if (item1.entityType) {
      return {
        text: item1.text,
        entityType: item1.entityType
      };
    } else {
      return {
        text: item1.text
      };
    }
  });
  return {
    type: "TYPE_EXAMPLE",
    parts
  };
})
```

Seems like the entityType logic can be merged. So you can make the code even more compact
```javascript
utterance.trainingPhrases = responses[0].trainingPhrases.map(item => ({
    type: "TYPE_EXAMPLE",
    parts: item.parts.map(item1 => ({
      text: item1.text,
      entityType: item1.entityType  
    }))
  })
);
```

Reply:
If you think you don't want entityType exists in some case, you can do like this:

```javascript
utterance.trainingPhrases = responses[0].trainingPhrases.map(item => ({
    type: "TYPE_EXAMPLE",
    parts: item.parts.map(item1 => Object.assign({ text: item1.text }, 
      item1.entityType ? { entityType: item1.entityType } : {} )
    )
  })
);
```